### PR TITLE
CSS: Add a support test for the hack for .css('marginRight') etc.

### PR DESCRIPTION
**Pull request by mzgol**
This hack turns out to be needed by Android 4.0-4.3.

Add a support test so that the hack is invoked only where needed.

Refs 3747cc642a48d2a5a8ac83069f66bddd33bea301

Refs gh-1815
Refs gh-1820
Refs gh-1842

### CODE REVIEW

**Comment by mzgol** [comment]
This is a `compat` version of PR #1842.

@gibson042 I'm not sure what to do with all these `// Minified:` comments, especially that this whole file will most likely need a refactor to save some size.

I'm submitting a PR instead of just landing because of doubts related to these comments.

**Comment by mzgol** [comment]
@gibson042 This was added by you in https://github.com/jquery/sizzle/commit/36973e9340b9e072a2077125cfc442dc6a42b926; could we test for `document.documentMode` instead of `addEventListener`?

It'd be good to make sure it's fixed in Edge so that we don't have to apply an even more hacky workaround soon...

**Comment by gibson042** [comment]
Just delete the affected "Minified" line(s) and we'll clean up the whole file later. Hopefully this doesn't add too much size, though.

**Comment by mzgol** [comment]
> Hopefully this doesn't add too much size, though.

This used to add 86 bytes, I refactored it a little and now it's +74. Note, though, that it's just the test, the workaround was already present.

On `master` the workaround + support test combo added 127 bytes. Quite a lot.

**Comment by mzgol** [comment]
PR updated. Down to +45 bytes.

File: src/css/support.js [hunk 1]
```diff
@@ -94,9 +98,10 @@ define([
 
 		// Check for getComputedStyle so that this code is not run in IE<9.
 		if ( window.getComputedStyle ) {
-			pixelPositionVal = ( window.getComputedStyle( div, null ) || {} ).top !== "1%";
-			boxSizingReliableVal =
-				( window.getComputedStyle( div, null ) || { width: "4px" } ).width === "4px";
+			divStyle = window.getComputedStyle( div, null );
+			pixelPositionVal = ( divStyle || {} ).top !== "1%";
+			boxSizingReliableVal = ( divStyle || { height: "4px" } ).height === "4px";
```
Comment (mzgol) [inline src/css/support.js]: Who does remember why we need these `||`?

cc @jquery/core

Comment (gibson042) [inline src/css/support.js]: Hidden I frames in Firefox 4, of course: http://bugs.jquery.com/ticket/8763

I'm thinking it's safe to remove them now. :wink:

**Pull request merged**